src/panels/lovelace/editor/hui-edit-card.ts

```diff
@@ -0,0 +1,369 @@
+  private _hass?: HomeAssistant;
+  private _cardId?: string;
+  private _originalConfig?: LovelaceConfig;
+  private _configElement?: LovelaceCardEditor | null;
+  private _uiEditor?: boolean;
+  private _configValue?: ConfigValue;
+  private _configState?: string;
+  private _saving: boolean;
+  private _loading: boolean;
+  private _isToggleAvailable?: boolean;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _cardId: {},
+      _originalConfig: {},
+      _configElement: {},
+      _configValue: {},
+      _configState: {},
+      _uiEditor: {},
+      _saving: {},
+      _loading: {},
+      _isToggleAvailable: {},
+    };
+  }
+
+  protected constructor() {
+    super();
+    this._loading = true;
+    this._saving = false;
+  }
+
+  set cardConfig(cardConfig: LovelaceConfig) {
+    this._originalConfig = cardConfig;
+    if (String(cardConfig.id) !== this._cardId) {
+      this._loading = true;
+      this._uiEditor = true;
+      this._configElement = undefined;
+      this._configValue = { format: "yaml", value: undefined };
+      this._configState = "OK";
+      this._isToggleAvailable = false;
+      this._cardId = String(cardConfig.id);
+      this._loadConfigElement();
+    }
+  }
+
+  public async openDialog(): Promise<void> {
+    // Wait till dialog is rendered.
+    await this.updateComplete;
+    this._dialog.open();
+  }
+
+  private get _dialog(): PaperDialogElement {
+    return this.shadowRoot!.querySelector("paper-dialog")!;
+  }
+
+  private get _previewEl(): HuiCardPreview {
+    return this.shadowRoot!.querySelector("hui-card-preview")!;
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <style>
+        paper-dialog {
+          width: 650px;
+        }
+        .center {
+          margin-left: auto;
+          margin-right: auto;
+        }
+        paper-button paper-spinner {
+          width: 14px;
+          height: 14px;
+          margin-right: 20px;
+        }
+        paper-spinner {
+          display: none;
+        }
+        paper-spinner[active] {
+          display: block;
+        }
+        .hidden {
+          display: none;
+        }
+        .element-editor {
+          margin-bottom: 16px;
+        }
+      </style>
+      <paper-dialog with-backdrop>
+        <h2>${this.localize("ui.panel.lovelace.editor.edit.header")}</h2>
+        <paper-spinner
+          ?active="${this._loading}"
+          alt="Loading"
+          class="center"
+          style="margin-bottom: 24px;"
+        ></paper-spinner>
+        <paper-dialog-scrollable
+          class="${classMap({ hidden: this._loading! })}"
+        >
+          ${
+            this._uiEditor && this._configElement !== null
+              ? html`
+                  <div class="element-editor">
+                    ${
+                      when(
+                        this._configElement,
+                        () => this._configElement,
+                        () => html`
```

Comment:
I think that you can just use the string `"loading…"`

Reply:
It's only important that the template returned from `render` is wrapped in `html`